### PR TITLE
Update test code

### PR DESCRIPTION
**Pull request by troydai**
1. Move fake logger and logger factory to TestUtility so as to move test
utilitys from OmniSharp.Tests to separate common codes and tests;
2. Remove NRefactory dependency
3. Update tests accordingly
4. Improve readability

NRefactory is terminated http://mikemdblog.blogspot.com/2015/07/the-end-of-nrefactory-6.html

### CODE REVIEW

File: tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs [hunk 1]
```diff
@@ -207,60 +216,65 @@ private string TrimLines(string source)
...
+        private async Task<RunCodeActionResponse> RunRefactoring(
+            string source,
+            string refactoringName,
+            bool wantsChanges = false)
         {
-            var refactorings = await FindRefactoringsAsync(source);
+            IEnumerable<OmniSharpCodeAction> refactorings = await FindRefactoringsAsync(source);
+            var names = refactorings.Select(r => r.Name).ToArray();
+            
             Assert.Contains(refactoringName, refactorings.Select(a => a.Name));
```
Comment (DustinCampbell) [inline tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs]: Maybe just use `names` here?

File: tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs [hunk 2]
```diff
@@ -207,60 +216,65 @@ private string TrimLines(string source)
...
 
-        private async Task<RunCodeActionResponse> RunRefactoring(string source, string refactoringName, bool wantsChanges = false)
+        private async Task<RunCodeActionResponse> RunRefactoring(
+            string source,
+            string refactoringName,
+            bool wantsChanges = false)
         {
-            var refactorings = await FindRefactoringsAsync(source);
+            IEnumerable<OmniSharpCodeAction> refactorings = await FindRefactoringsAsync(source);
+            var names = refactorings.Select(r => r.Name).ToArray();
```
Comment (DustinCampbell) [inline tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs]: `names` seems to be unused?

**Comment by DustinCampbell** [comment]
I'm totally in favor of this change. :+1:

**Comment by DustinCampbell** [comment]
LGTM

File: tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs (hunk 1 above)
Comment (troydai) [inline tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs]: will do.

File: tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs (hunk 2 above)
Comment (troydai) [inline tests/OmniSharp.Roslyn.CSharp.Tests/CodeActionsV2Facts.cs]: oh, yah. just used for testing. will remove.

**Comment by troydai** [comment]
@DustinCampbell since "sort using" is removed as code action. Does it has a replacement in Roslyn?

**Comment by willl** [comment]
I recall that there was effort to move from NRefactory to RefactoringEssentials as part of the move to beta5 or 8? It never got completed due to various version incompatibilities at the time.

Here's the issue that was posted at the time https://github.com/icsharpcode/RefactoringEssentials/issues/40

**Comment by DustinCampbell** [comment]
Not at the moment. We haven't added an analyzer, fix, or refactoring for sort usings. 😦

**Comment by troydai** [comment]
need a lgtm, again? not familiar with the approvals/lgtm.

**Pull request merged**